compiler/noirc_frontend/src/hir_def/types.rs

```diff
@@ -2746,7 +2746,13 @@ impl From<&Type> for PrintableType {
                     .expect("Cannot print variable sized strings");
                 PrintableType::String { length: size }
             }
-            Type::FmtString(_, _) => unreachable!("format strings cannot be printed"),
+            Type::FmtString(size, typ) => {
+                let dummy_location = Location::dummy();
+                let size = size
+                    .evaluate_to_u32(dummy_location)
+                    .expect("Cannot print variable sized strings");
```

Comment:
Maybe we should issue an actual error here instead of moving the panic to a less common situation?
Although the error message would use the dummy location it seems. I wonder if we can forward the location from the println somehow.

Reply:
Changing this would require some form of Result, and at least in one part of the code there's no Result returned from a function that calls this, so...
Before making this change, it would be nice to find a way to trigger this failure. A similar panic exists for the string case but I don't know if it's possible to end up with `str<N>` or `fmtstr<N, T>` where `N` is not a constant.

Reply:
> A similar panic exists for the string case but I don't know if it's possible to end up with `str<N>` or `fmtstr<N, T>` where `N` is not a constant.

Strings are treated as arrays so `N` should always be constant. We do the same things for strings, moving away from this panic feels like a separate issue to this PR.